### ISSUE TITLE
Search results page: Page should contain a level-one heading

### ISSUE THREAD

**Issue by lhenze** [inveniosoftware/invenio-app-rdm#3039]
Package version (if known): v12.0.5 

## Describe the bug
For accessibility best practices, every page should have a level one heading, ideally marked up as `<h1>`

## Steps to Reproduce
1. Go to a ['search results page'](https://inveniordm.web.cern.ch/search?q=&l=list&p=1&s=10&sort=newest)
2. View source
3. Note that there is no level one header

## Expected behavior

The markup would contain an `<h1>` that says "Search results" or "Search results: 1,600 items" or other summary of the contents of the page.  If visible, this change would also clarify the visual hierarchy of the page for sighted users.  If it is preferred not to be visible, the `<h1>` tag can be hidden to all but screenreaders.  

## Screenshots - current

![Image](https://github.com/user-attachments/assets/c1846dbf-46d3-4585-9283-a4194859997b)

## Screenshots - proposed

![Image](https://github.com/user-attachments/assets/f7a8e236-eb7d-4891-9472-9533982c82d0)

![Image](https://github.com/user-attachments/assets/a2e99f96-2698-4ab2-a1cf-0ff1eab5e2e3)

## Additional context

This was mentioned in the NYU accessibility review a few years back and has continued to pop up when I run automated tests. 

https://dequeuniversity.com/rules/axe/4.10/page-has-heading-one
https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html

**Comment by lhenze** [comment]
@kpsherva @zzacharo
